### DISCUSSION TITLE
Tailwind Inline color decorators are not working with CSS variables

### DISCUSSION THREAD

**Discussion by RowinVanAmsterdam** [tailwindlabs/tailwindcss#12225] [Help] [answered]
Using CSS variables for multi-theme support in tailwind require you to define colors in rgb, hsl or rgba format, without the color space function:

Works:
`  --color-primary: 255 115 179;`
Won't work:
` --color-primary: rgb(255 115 179);`

Like described in the official docs: https://tailwindcss.com/docs/customizing-colors#using-css-variables

This means that you'll lose the inline color decorators, and also IDE's like vscode won't show any preview of the color (see screenshot, no preview before `bg-secondary-100`). Which makes it pretty hard to recognise the assigned color.  

<img width="372" alt="Screenshot 2023-10-17 at 21 23 40" src="https://github.com/tailwindlabs/tailwindcss/assets/33234293/7316bd8d-50a2-493a-9629-6b24864d2460">

I know you could assign the hex as css variable directly, but this means you'll lose the option to use an opacity, for example `bg-secondary-100/10` won't work. 

Is there a way to keep inline decorator as well as the opacity functionality?  

----------

My current project setup:
```
// themes.css

@layer base {
    :root,
    .theme-light {
        /* THEME COLORS */
        --color-primary-50: 8 67 103;
        --color-primary-100: 49 49 49;
        --color-primary-150: 28 28 28;
```
        
``` 
// tailwind.config.ts

    theme: {
        extend: {
            colors: {
                primary: {
                    50: 'rgb(var(--color-primary-50) / <alpha-value>)',
                    100: 'rgb(var(--color-primary-100) / <alpha-value>)',
                    150: 'rgb(var(--color-primary-150) / <alpha-value>)'
                },
```

**Comment by wongjn** [comment, marked as answer]
I wouldn't think there is way to keep inline decorator, since the value of the CSS variable could be sourced from practically anywhere, from the point of view of the VSCode extension. Furthermore, CSS variable values can change at runtime too so the inline decorator could be inaccurate, even it if it *was* able to somehow resolve the CSS variable value.

**Reply by RowinVanAmsterdam** [reply]
Hmm that made sense, because the value of a CSS variable can change of course. So there is no way to know what the actual value is. Shoot I didn't think about this, but it's still sad. The preview makes assigning/debugging colors much easier.

**Reply by castortech** [reply]
Wouldn't there be a way to specify for example light or dark theme or a set value that could be used, understanding that it represent but one possible final value?

**Reply by RowinVanAmsterdam** [reply]
> Wouldn't there be a way to specify for example light or dark theme or a set value that could be used, understanding that it represent but one possible final value?

Not natively, since it's in your IDE and there is no support for that afaik.

**Reply by Ganbin** [reply]
@wongjn I understand the answer for when we use the color in our code. However do you think it would be possible to still show the inline color decorators in the css file where the variables are defined. We would need to "guess" the type of function we implicitly want to use, but I suppose (didn't try out) some regex could do the trick.
